securedrop_client/utils.py

```diff
@@ -85,3 +89,21 @@ def chronometer(logger: logging.Logger, description: str) -> Generator:
     finally:
         elapsed = time.perf_counter() - start
         logger.debug(f"{description} duration: {elapsed:.4f}s")
+
+
+class SourceCache(object):
+    """
+    Caches Sources by UUID.
+    """
+    cache: Dict[str, db.Source] = {}
+    session = None
+
+    def __init__(self, session: Session) -> None:
+        super().__init__()
+        self.session = session
+
+    def get(self, source_uuid: str) -> Optional[db.Source]:
+        if source_uuid not in self.cache:
+            source = self.session.query(db.Source).filter_by(uuid=source_uuid).first()
```

Comment:
Confirming that using `first` is better than using `one` because it returns `None` instead of throwing `NoResultFound`.